### DISCUSSION TITLE
Mapping No working.

### DISCUSSION THREAD

**Discussion by mhmetalheart** [darthcloud/BlueRetro#1095] [General]
### BlueRetro firmware version

v24.10_hw1

### BlueRetro firmware specification

HW1

### BlueRetro firmware variant

System specific

### BlueRetro hardware type

Internal install DIY

### Manufacturer

DIY

### System used

Sony PlayStation 2

### Bluetooth controller brand & name

sony PS3 PS4 & 8 bitdo SN30 pro M30

### What is problem? (only list ONE problem per report)

in mapping. I set different 'device 2 (or 3,4...)'
For example: Device 2. src=M30, dst=PSX. set A=X(default is square), B to O (default is X). Save. and I can see the green notice.
Then, connect my M30, Nothing happened. The M30 still in default setting which the A=square, B=X,C=O.
Also tried my humble bazooka's Saturn adapter, same issue (see attach). I set B to A, and A to B. But, nothing happened. A still A. and B still B.
My friend who using the retrotime's DC adapter also got the same issue. 
Another friend who use DIY NGC receiver, said that mapper setting is work. He will test again tomorrow.

Maybe I did something wrong?

### What did you expect to happen?

The mapping device setting should be work, right?

### Attach files like logs or Bluetooth traces here

![humblebazooka](https://github.com/user-attachments/assets/b6f9079a-779c-4147-b131-2d7c48af0b98)

**Comment by darthcloud** [comment]
Thats not how mapping work.

Mapping are not tied to a particular controller.

Device 1 means player 1, device 2 means player 2.

The label are only to help map the button they have no function.

**Comment by mhmetalheart** [comment]
thanks. now I understand.
